### PR TITLE
Automatic processing (simplified UI)

### LINKED ISSUE #3093: [NLP] Automated Transcription Modal
## Description

Allows to choose a service to transcribe audio with.

Possibly would be able to work on this without Payments #3101 as services probably offer free limited services (to be checked)

## Additional details

Requires #3099
Reference mockups in [this 2022 Figma file](https://www.figma.com/file/Mk2DB7adukIZiYpyQlsKps/NLP-Mockups-2022?node-id=0%3A1)

### LINKED ISSUE #3104: [NLP] Automated translation modal
## Description

Allows to choose a service to translate the transcribed text.

Possibly would be able to work on this without Payments #3101 as services probably offer free limited services (to be checked)

## Additional details

Requires #3099
Reference mockups in [this 2022 Figma file](https://www.figma.com/file/Mk2DB7adukIZiYpyQlsKps/NLP-Mockups-2022?node-id=0%3A1)

### PR DESCRIPTION
**Pull request by magicznyleszek**
## Description

Enables automatic transcription and translation (via just one service: Google) using a simplified UI (a button with a spinner).

## Related issues

Closes #3104 
Closes #3093

### CODE REVIEW

**Review by duvld — COMMENTED** [review]
Frontend reviewed

File: jsapp/js/components/common/audioPlayer.tsx [hunk 1]
```diff
@@ -42,20 +46,22 @@ class AudioPlayer extends React.Component<AudioPlayerProps, AudioPlayerState> {
   }
 
   componentDidMount() {
-    this.prepareAudio();
+    // Prepare audio.
```
[the remark this answers is not in the document]
Comment (magicznyleszek) [reply jsapp/js/components/common/audioPlayer.tsx:49]: I was getting warnings/errors from React sometimes when switching routes that there is possible memory leak (as the `this.audioInterface.onX` were not being cancelled), so I just rewrote parts of it

File: jsapp/js/components/metadataEditor.es6 [hunk 2]
```diff
@@ -235,6 +235,19 @@ export default class MetadataEditor extends React.Component {
           </bem.FormBuilderMeta__column>
         </bem.FormBuilderMeta__columns>
 
+        {this.isAuditEnabled() && (
```
[the remark this answers is not in the document]
Comment (magicznyleszek) [reply jsapp/js/components/metadataEditor.es6]: I think it was merged already and looks ok?

File: jsapp/js/components/processing/processingActions.ts [hunk 3]
```diff
@@ -14,11 +14,31 @@ const NO_FEATURE_ERROR = t('Asset seems to not have the processing feature enabl
 // character as value.
 const DELETE_CHAR = '⌫';
 
+interface GoogleTsResponse {
+  status: 'complete';
+  /** Full transcript text. */
+  value: string;
+  /** Transcript text split into chunks - scored by transcription quality. */
```
[the remark this answers is not in the document]
Comment (magicznyleszek) [reply jsapp/js/components/processing/processingActions.ts:21]: We might display it in the UI at some point - so if the score is very low, highlight that part and notify user

File: jsapp/js/components/helpBubbles.es6 [hunk 4]
```diff
@@ -17,7 +17,8 @@ class HelpBubble extends React.Component {
     autoBind(this);
     this.state = {
       isOpen: false,
-      isOutsideCloseEnabled: true
+      isOutsideCloseEnabled: true,
+      locallyAcknowledgedMessageUids: new Set()
```
[the remark this answers is not in the document]
Comment (magicznyleszek) [reply jsapp/js/components/helpBubbles.es6]: I'm not sure why these changes appear here as changes, as they come from #3787 :shrug:

**Pull request merged**